modules/overview-of-data-science-pipelines-caching.adoc

```diff
@@ -0,0 +1,70 @@
+= Overview of data science pipelines caching
+
+[role='_abstract']
+{productname-short} supports caching within data science pipelines to optimize execution times and improve resource efficiency. Using caching reduces redundant task execution by reusing results from previous runs with identical inputs. 
+
+Caching is particularly beneficial for iterative tasks, where intermediate steps might not need to be repeated. Understanding caching can help you design more efficient pipelines and save time in model development.
+
+Caching operates by storing the outputs of successfully completed tasks and comparing the inputs of new tasks against previously cached ones. If a match is found, {productname-short} reuses the cached results instead of re-executing the task, reducing computation time and resource usage.
+
+== Caching criteria
+
+For caching to be effective, the following criteria determines if a task can use previously cached results:
+
+* *Input data and parameters*: If the input data and parameters for a task are unchanged from a previous run, cached results are eligible for reuse.
+* *Task code and configuration*: Changes to the task code or configurations invalidate the cache to ensure that modifications are always reflected.
+* *Pipeline environment*: Changes to the pipeline environment, such as dependency versions, also affect caching eligibility to maintain consistency.
+
+== Viewing cached steps in the {productname-short} user interface
+
+Cached steps in pipelines are visually indicated in the user interface (UI):
+
+* Tasks that use cached results display a green icon, helping you quickly identify which steps were cached.
+* The UI also includes information about when the task was previously executed, allowing for easy verification of cache usage.
+
+To confirm caching status for specific tasks, navigate to the pipeline details view in the UI, where all cached and non-cached tasks are indicated.
+
+== Disabling caching for specific tasks or pipelines
+
+In {productname-short}, caching is enabled by default, but there are situations where disabling caching for specific tasks or the entire pipeline is necessary. For example, tasks that rely on frequently updated data or unique computational needs might not benefit from caching.
+
+=== Disabling caching for individual tasks
+
+To disable caching for a particular task, apply the `set_caching_options` method directly to the task in your pipeline code:
+
+`task_name.set_caching_options(False)`
+
+After applying this setting, {productname-short} executes the task in all future pipeline runs, ignoring any cached results.
+
+[NOTE]
+====
+You can re-enable caching for individual tasks by setting `set_caching_options(True)`.
+====
+
+=== Disabling caching for pipelines
+
+If necessary, you can disable caching for the entire pipeline during pipeline submission by setting the `enable_caching` parameter to `False`. This setting ensures that no steps are cached during pipeline execution.
```

Comment:
The _enable_caching_ parameter is only available when using the [kfp.client](https://kubeflow-pipelines.readthedocs.io/en/sdk-2.9.0/source/client.html#) to summit pipelines or starting pipeline runs, like in the [run_pipeline](https://kubeflow-pipelines.readthedocs.io/en/sdk-2.9.0/source/client.html#kfp.client.Client.run_pipeline.enable_caching) method

I think it is important to remark this, as the UI doesn't have any enable_pipeline parameter and neither can be set when creating a python pipeline

Reply:
@aduquett - I will update the text with Jorge's suggestion.

Reply:
Updated the text with Jorge's proposed content.